lib/core/api/api-factory.js

```diff
@@ -132,7 +132,7 @@
             // if service has a callback then call it
             // var response = self._createResponse(data, status, headers, _config);
             if(afterCallback) {
-              successResponse = afterCallback.call(self, successResponse);
+              successResponse = afterCallback.call(self, successResponse, config.data);
```

Comment:
This is to support the use case of needing to update the local model with the id returned by the create API.